src/components/ManageAssets/ManageAssetsDropdown/ManageAssetsDropdown.tsx

```diff
@@ -0,0 +1,113 @@
+import React from "react";
+import { useDispatch } from "react-redux";
+import styled from "@emotion/styled";
+import { mixin, vars } from "../../../styles";
+import { Dropdown } from "react-bootstrap";
+import { loadAdminAddress } from "../../../reducers/admin";
+import { connectToMetamask } from "../../../services/etherjs";
+
+interface ManageAssetsDropdownProps {
+  userWalletAddress: string;
+  approvedBeneficiaryAddress: string;
+  approvedHolderAddress: string;
+  beneficiaryAddress: string;
+  holderAddress: string;
+  metamaskAccountError: string;
+  className?: string;
+}
+
+export const ManageAssetsDropdown = ({
+  userWalletAddress,
+  approvedBeneficiaryAddress,
+  approvedHolderAddress,
+  beneficiaryAddress,
+  holderAddress,
+  metamaskAccountError,
+  className,
+}: ManageAssetsDropdownProps) => {
+  const dispatch = useDispatch();
+
+  const isApprovedAddressPending = !!approvedBeneficiaryAddress && !!approvedHolderAddress;
```

Comment:
isApprovedAddressPending is bit confusing. it seems it shows that data is still getting fetched. rather put simply as haveApprovedAddressess or something else.